### PR TITLE
Add game registry and simple wrapper for game state management

### PR DESCRIPTION
**Pull request by alexcavalli**
@denheck I'm not especially attached to this particular game state structure, it's probably a bit more complex than we'll end up actually needing. We could probably get away with using the names provided by the players as their ids. (maybe. we'd have to decide if we triggered the join game/add player behavior when the player arrives at the game page, or after they have input a player name. if the former, we may want to just generate an id for them and let them set a name to go along with that id later)

### CODE REVIEW

File: lib/wikitrivia/game.ex [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    game_id = Ecto.UUID.generate
+    {:ok, _} = Agent.start_link(fn -> default_state end, name: agent_name_by_game_id(game_id))
+    game_id
+  end
+
+  def get_game_state(game_id) do
+    Agent.get(agent_name_by_game_id(game_id), fn s -> s end)
+  end
+
+  def add_player(game_id, player_id, player_name) do
```
Comment (denheck) [inline lib/wikitrivia/game.ex]: @alexcavalli I like your idea of having the player name as the ID. That would also help ensure names are unique.

**Review by denheck — APPROVED** [review]
👍 Looks good to me

File: lib/wikitrivia/game.ex (hunk 1 above)
Comment (alexcavalli) [reply lib/wikitrivia/game.ex]: @denheck So, then, an initial structure maybe like this?
```elixir
%{
  players: ["player_1_name", etc] # MapSet or List or something, depending on if we want to retain order
  scores: %{
    "player_1_name" => 10,
    etc.
  } 
}
```

**Comment by alexcavalli** [comment]
@denheck Changed the game state a bit, using the player names as ids, as you suggested.

**Pull request merged**